cloudferrylib/os/compute/nova_compute.py

```diff
@@ -583,6 +583,27 @@ def attach_volume_to_instance(self, instance, volume):
     def dissociate_floatingip(self, instance_id, floatingip):
         self.nova_client.servers.remove_floating_ip(instance_id, floatingip)
 
+    def get_hypervisor_statistics(self):
+        return self.nova_client.hypervisors.statistics()
+
     def get_hypervisors(self):
         return [hypervisor.host_ip for hypervisor in
                 self.nova_client.hypervisors.list()]
+
+    def get_free_vcpus(self):
+        hypervisor_statistics = self.get_hypervisor_statistics()
+        return hypervisor_statistics.vcpus * float(
```

Comment:
I guess here we will not need type conversions, if have it defined in config options.